### PR TITLE
Created Spell class

### PR DESCRIPTION
**Pull request by rightonhana**
Created Spell class

### CODE REVIEW

File: src/Spell.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
+import java.util.Objects;
+
+public class Spell {
```
Comment (AlejandroWoloch) [inline src/Spell.java]: this has to implement the Ability interface

Comment (rightonhana) [reply src/Spell.java]: Sorry! I forgot about that

File: src/Spell.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
+public class Spell extends Ability {
```
Comment (ducampcarlos) [inline src/Spell.java]: Spell should be Abstract

File: src/Spell.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+	// Attributes
+	private static int lastId = 0;
```
Comment (AlejandroWoloch) [inline src/Spell.java]: Ability class has lastId already

File: src/Spell.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+	// Override toString method.
+	@Override
+	public String toString() {
+		String base = super.toString();
+		String messageFmt = "\n=> Spell\n[Damage: %d]";
+		return base.concat(String.format(messageFmt, getDamage()));
+	}
+	
+	@Override
+	public byte calculateCost() {
```
Comment (ducampcarlos) [inline src/Spell.java]: This should be implemented in the concrete classes

File: src/Spell.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+	// Constructors
+	public Spell(int damage, String name, String classUser) throws Exception {
+		super(name, classUser);
+		this.setId();
+		this.setDamage(damage);
+	}
+	
+	public Spell(String name, byte level, String classUser, int damage) throws Exception {
+		super(name, level, classUser);
+		this.setId();
```
Comment (ducampcarlos) [inline src/Spell.java]: ID is setted in the super class constructor

**Comment by ducampcarlos** [comment]
Looks ok

**Comment by AlejandroWoloch** [comment]
It's okey

**Pull request merged**